### PR TITLE
fix(MapValidator): Validation of objects with dot notation

### PR DESCRIPTION
**Pull request by martinalbert**
## Description

Refactors handling variables that are written with dot notation, e.g. `output.is.nested = true`. 

## Motivation and Context

Wrong linter errors.

```
Map for profile: "address/geocoding" and provider: "nominatim" found on local filesystem

🆗 Parsing map file: /station/capabilities/address/geocoding/maps/nominatim.suma

⚠️ Validating profile: /station/capabilities/address/geocoding/profile.supr to map: /station/capabilities/address/geocoding/maps/nominatim.suma
ObjectLiteral - Wrong Object Structure: expected addressCountry, addressRegion, addressLocality, streetAddress, postalCode, formattedAddress, but got , , , , , , , , , , , 

```

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTION_GUIDE** document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by lukas-valenta — CHANGES_REQUESTED** [review]
looks good, fix the tests and it's good to go

**Review by lukas-valenta — APPROVED** [review]

File: src/interpreter/map-validator.test.ts [hunk 1]
```diff
@@ -577,11 +676,68 @@ describe('MapValidator', () => {
...
+            }`
+          );
+
+          validWithWarnings(profileAst, [mapAst1]);
+          invalidWithErrors(
+            profileAst,
+            [mapAst2],
+            `ObjectLiteral - Wrong Structure: expected number, but got "ObjectLiteral"
+ObjectLiteral - Wrong Structure: expected number, but got "ObjectLiteral"
+15:34 PrimitiveLiteral - Wrong Structure: expected number, but got "false"
```
[the remark this answers is not in the document]
Comment (lukas-valenta) [reply src/interpreter/map-validator.test.ts:708]: i see now it's all one string, should be array of strings imo

[the remark this answers is not in the document]
Comment (martinalbert) [reply src/interpreter/map-validator.test.ts:708]: True, I'll let you know when it'll be ready

[the remark this answers is not in the document]
Comment (martinalbert) [reply src/interpreter/map-validator.test.ts]: @lukas-valenta check it out pls

**Review by lukas-valenta — APPROVED** [review]
Looks great :+1:

**Review by lukas-valenta — APPROVED** [review]

**Pull request merged**